core/services/pg/q.go

```diff
@@ -280,10 +282,28 @@ func sprintQ(query string, args []interface{}) string {
 	}
 	var pairs []string
 	for i, arg := range args {
-		pairs = append(pairs, fmt.Sprintf("$%d", i+1), fmt.Sprintf("%v", arg))
+		// We print by type so one can directly take the logged query string and execute it manually in pg.
+		// Annoyingly it seems as though the logger itself will add an extra \, so you still have to remove that.
+		switch v := arg.(type) {
+		case []byte:
+			pairs = append(pairs, fmt.Sprintf("$%d", i+1), fmt.Sprintf("'\\x%x'", v))
+		case common.Address:
+			pairs = append(pairs, fmt.Sprintf("$%d", i+1), fmt.Sprintf("'\\x%x'", v.Bytes()))
+		case common.Hash:
+			pairs = append(pairs, fmt.Sprintf("$%d", i+1), fmt.Sprintf("'\\x%x'", v.Bytes()))
+		case pq.ByteaArray:
+			s := fmt.Sprintf("('\\x%x'", v[0])
+			for j := 1; j < len(v); j++ {
+				s += fmt.Sprintf(",'\\x%x'", v[j])
+			}
```

Comment:
Worth a [`strings.Builder`](https://pkg.go.dev/strings#Builder) to minimize allocs?
```suggestion
			var s strings.Builder
			fmt.Fprintf(&s, "('\\x%x'", v[0])
			for j := 1; j < len(v); j++ {
				fmt.Fprintf(&s, ",'\\x%x'", v[j])
			}
```

Reply:
Might need to pass a `&s` or call `String()` below too :facepalm: